US-002

thundergod-vicky opened the pull request:
Refactor routing and authentication in App component

- Introduced ProtectedRoute and PublicRoute components to manage access control for routes.

- Updated App component to utilize new routing structure with nested routes for public and protected access.

- Simplified the Header component by removing unused authentication logic and improving menu toggle functionality.

- Enhanced AppShowcase component with a direct navigation link for "Learn More".

- Cleaned up imports and removed commented-out code for better readability.

acceptance criteria 

-All the links are visible

-All the 4 links are working at the footer section

-The sublinks are working inside the links

-On Refresh everytime it  redirecting to dashboard/home based on logged in or out

Pritammondal001 commented:
Acceptance criteria
Criteria 1- **Passed**
Criteria 2- **Passed**
Criteria 3- **Passed**
Criteria 4- **Passed**

tatz210588 reviewed (changes requested):
Edit your code and acceptance criteria. Completing a US, should not damage another US. 
1) 1st Loggin in should redirect to Dashboard. 
2) Without Login, Reward page should show a message for "Login Now".
This system is already existing in production, but was removed in this PR.

thundergod-vicky commented:
Refactor routing and authentication in App component

Introduced ProtectedRoute and PublicRoute components to manage access control for routes.

Updated App component to utilize new routing structure with nested routes for public and protected access.

Simplified the Header component by removing unused authentication logic and improving menu toggle functionality.

Enhanced AppShowcase component with a direct navigation link for "Learn More".

Cleaned up imports and removed commented-out code for better readability.

acceptance criteria

-All the links are visible

-All the 4 links are working at the footer section

-The sublinks are working inside the links

-On Refresh everytime it redirecting to dashboard/home based on logged in or out

(UPDATED)

Pritammondal001 commented:
Acceptance criteria
Criteria 1- Passed
Criteria 2- Passed
Criteria 3- Passed
Criteria 4- Passed

tatz210588 commented:
There are 5 Acceptance criteria for this US as per Scrum master. However, the tester passed only 4.

Pritammondal001 commented:
Acceptance criteria
Criteria 1- Passed
Criteria 2- Passed
Criteria 3- Passed
Criteria 4- Passed
Criteria 5- Passed
Updated

tatz210588 reviewed (approved):
Ok
